user_management/api/views.py

```diff
@@ -151,7 +152,8 @@ def initial(self, request, *args, **kwargs):
         try:
             self.user = User.objects.get(pk=uid)
         except User.DoesNotExist:
-            raise Http404()
+            msg = _('Invalid or expired token.')
+            raise PermissionDenied(detail=msg)
```

Comment:
Meshy's Law of HTTP status codes:

As an HTTP status code discussion grows longer, the probability of a comparison involving Teapots approaches 1.